ocrd/ocrd/cli/workspace.py

```diff
@@ -229,42 +229,57 @@ def workspace_add_file(ctx, file_grp, file_id, mimetype, page_id, ignore, check_
     workspace.save_mets()
 
 # ----------------------------------------------------------------------
-# ocrd workspace add-bulk
+# ocrd workspace bulk-add
 # ----------------------------------------------------------------------
 
 # pylint: disable=broad-except
 @workspace_cli.command('bulk-add')
 @click.option('-r', '--regex', help="Regular expression matching the FILE_GLOB filesystem paths to define named captures usable in the other parameters", required=True)
 @click.option('-m', '--mimetype', help="Media type of the file. If not provided, guess from filename", required=False)
 @click.option('-g', '--page-id', help="physical page ID of the file", required=False)
-@click.option('-i', '--file-id', help="ID of the file", required=True)
-@click.option('-u', '--url', help="local filesystem path in the workspace directory (copied from source file if different)", required=True)
+@click.option('-i', '--file-id', help="ID of the file", required=False)
+@click.option('-u', '--url', help="local filesystem path in the workspace directory (copied from source file if different)", required=False)
 @click.option('-G', '--file-grp', help="File group USE of the file", required=True)
 @click.option('-n', '--dry-run', help="Don't actually do anything to the METS or filesystem, just preview", default=False, is_flag=True)
+@click.option('-S', '--source-path', help="File path to copy from", default='{{ src }}')
 @click.option('-I', '--ignore', help="Disable checking for existing file entries (faster)", default=False, is_flag=True)
 @click.option('-f', '--force', help="Replace existing file entries with the same ID (no effect when --ignore is set, too)", default=False, is_flag=True)
 @click.option('-s', '--skip', help="Skip files not matching --regex (instead of failing)", default=False, is_flag=True)
 @click.argument('file_glob', nargs=-1, required=True)
 @pass_workspace
-def workspace_cli_bulk_add(ctx, regex, mimetype, page_id, file_id, url, file_grp, dry_run, file_glob, ignore, force, skip):
-    r"""
+def workspace_cli_bulk_add(ctx, regex, mimetype, page_id, file_id, url, file_grp, dry_run, file_glob, source_path, ignore, force, skip):
+    """
     Add files in bulk to an OCR-D workspace.
 
-    FILE_GLOB can either be a shell glob expression or a list of files.
+    FILE_GLOB can either be a shell glob expression or a list of files or '-'
+    in which case file paths are read from STDIN.
 
     --regex is applied to the absolute path of every file in FILE_GLOB and can
     define named groups that can be used in --page-id, --file-id, --mimetype, --url and
     --file-grp by referencing the named group 'grp' in the regex as '{{ grp }}'.
 
+    If the file paths aren't really file paths but CSV data that *contain* a
+    filename, the part that contains the actual filename should be in the named
+    group 'src'. If not explicitly defined, 'src' is equal to the file path.
```

Comment:
```suggestion
    If the FILE_GLOB expressions do not denote the file names themselves
    (but arbitrary strings for --regex matching), then use --source-path to set
    the actual file paths to use. (This could involve fixed strings or group references.)
```